apps/antalmanac/src/components/RightPane/SectionTable/SectionTable.tsx

```diff
@@ -64,12 +66,52 @@ const styles = {
     },
 };
 
+const TableHeadColumns: { value: string; label: string }[] = [
+    { value: 'sectionCode', label: 'Code' },
+    { value: 'sectionDetails', label: 'Type' },
+    { value: 'instructors', label: 'Instructors' },
+    { value: 'dayAndTime', label: 'Times' },
+    { value: 'location', label: 'Places' },
+    { value: 'sectionEnrollment', label: 'Enrollment' },
+    { value: 'restrictions', label: 'Rstr' },
+    { value: 'status', label: 'Status' },
+];
+
 const SectionTable = (props: SectionTableProps) => {
     const { classes, courseDetails, term, colorAndDelete, highlightAdded, scheduleNames, analyticsCategory } = props;
     const courseId = courseDetails.deptCode.replaceAll(' ', '') + courseDetails.courseNumber;
     const encodedDept = encodeURIComponent(courseDetails.deptCode);
     const isMobileScreen = useMediaQuery(`(max-width: ${MOBILE_BREAKPOINT}`);
 
+    const [columns, setColumns] = useState([
+        'sectionCode',
+        'sectionDetails',
+        'instructors',
+        'dayAndTime',
+        'location',
+        'sectionEnrollment',
+        'restrictions',
+        'status',
+    ]);
+
+    const [activeTableHeadColumns, setActiveTableHeadColumns] = useState(
+        TableHeadColumns.filter((column) => columns.includes(column.value))
+    );
+
+    useEffect(() => {
+        RightPaneStore.on('columnChange', (columns) => {
+            setColumns(columns);
+            setActiveTableHeadColumns(TableHeadColumns.filter((column) => columns.includes(column.value)));
+        });
+
+        return () => {
+            RightPaneStore.removeListener('columnChange', (columns) => {
+                setColumns(columns);
+                setActiveTableHeadColumns(TableHeadColumns.filter((column) => columns.includes(column.value)));
+            });
+        };
```

Comment:
Extract callback to `changeColumns()` or something along these lines so that you're not duplicating code